packages/platform/setup-accounts.js

```diff
@@ -21,22 +24,31 @@ export const buildContext = (user) => {
 };
 
 export default ({ mergeUserCartsOnLogin = true } = {}) => {
-  Accounts.validateNewUser((user) => {
+  accountsPassword.options.validateNewUser = (user) => {
     const clone = cloneDeep(user);
+    if (clone.email) {
+      clone.emails = [
+        {
+          address: clone.email,
+          verified: false,
+        },
+      ];
+      delete clone.email;
+    }
     delete clone._id;
-    Users.simpleSchema().validate(clone);
-    return true;
-  });
+    Users.simpleSchema()
+      .extend({
+        password: String,
+      })
+      .omit('created')
+      .validate(clone);
+    return user;
+  };
 
-  Accounts.onCreateUser((options = {}, user = {}) => {
+  accountsServer.on('CreateUserSuccess', async ({ user, options }) => {
```

Comment:
I think CreateUserSuccess is not the equivalent of Accounts.onCreateUser, maybe it has to be moved to  and I guess the google and facebook related code has to be removed as there is own strategies for that in accounts-js.

Returning a user from the event will not do anything, so changes to newUser will be going to nirvana.

And I'm not sure that skipEmailVerification option does what it should

Reply:
Also I think the closest thing to Accounts.onCreateUser is if you use this function here to adjust the user fields: https://www.accountsjs.com/docs/api/server/interfaces/accountsserveroptions#optional-userobjectsanitizer

Reply:
I've just seen the sanitizer function is only used for e-mail verifications but the CreateUserSuccess event is still likely the wrong place and facebook, google code obsolete

Reply:
(you can break login with facebook or login with google stuff)